### DISCUSSION TITLE
Need help updating pricing through UpdateInventory

### DISCUSSION THREAD

**Discussion by zkutty** [etsy/open-api#1075] [Q&A]
Hi there, I'm pretty new to using APIs but have been able to write some python code to pull out all sorts of inventory information and create a pricing recommendation by SKU based on my desired margin profiles.  I now want to update my inventory in Etsy to reflect this new pricing, but can't get past an error which says: `{"error":"Missing input parameter: [products]"}\n` when trying to upload.

I'll drop in the code I'm using below, but used the GetInventory endpoint to pull out the inventory listing for a listing_id and all I did was change the SKU pricing and now want to upload, so not sure why its saying my payload is malformed?

I'm sure its something quick and simple but would really appreciate some help!

upload code:
`def update_listing_inventory(listing_id:int, data:dict, token, keystring):
    import json
    from requests_oauthlib import OAuth2Session
    headers = {
		"Accept": "application/json",
		"Content-Type": "application/x-www-form-urlencoded",
		"x-api-key": keystring,
    	      }

    refresh_url = "https://api.etsy.com/v3/public/oauth/token"

    etsy_auth = OAuth2Session(keystring, token=token, auto_refresh_url=refresh_url)

    r = etsy_auth.put(f'https://openapi.etsy.com/v3/application/listings/{listing_id}/inventory', headers=headers, data = data)
    return r.text`

data to upload:
`{"products": [{"product_id": 17215385272, "sku": "14782886189767442012", "is_deleted": false, "offerings": [{"offering_id": 17178103942, "quantity": 999, "is_enabled": true, "is_deleted": false, "price": {"amount": 20.99, "divisor": 100, "currency_code": "USD"}}], "property_values": [{"property_id": 513, "property_name": "Colors", "scale_id": null, "scale_name": null, "value_ids": [50135267836], "values": ["Black"]}, {"property_id": 514, "property_name": "Sizes", "scale_id": null, "scale_name": null, "value_ids": [54657375794], "values": ["S"]}]}, {"product_id": 16865398421, "sku": "24152318356747733749", "is_deleted": false, "offerings": [{"offering_id": 17178103944, "quantity": 999, "is_enabled": true, "is_deleted": false, "price": {"amount": 20.99, "divisor": 100, "currency_code": "USD"}}], "property_values": [{"property_id": 513, "property_name": "Colors", "scale_id": null, "scale_name": null, "value_ids": [50135267836], "values": ["Black"]}, {"property_id": 514, "property_name": "Sizes", "scale_id": null, "scale_name": null, "value_ids": [56802856687], "values": ["M"]}]}, {"product_id": 17215385274, "sku": "24574928614471245926", "is_deleted": false, "offerings": [{"offering_id": 16982444301, "quantity": 999, "is_enabled": true, "is_deleted": false, "price": {"amount": 20.99, "divisor": 100, "currency_code": "USD"}}], "property_values": [{"property_id": 513, "property_name": "Colors", "scale_id": null, "scale_name": null, "value_ids": [50135267836], "values": ["Black"]}, {"property_id": 514, "property_name": "Sizes", "scale_id": null, "scale_name": null, "value_ids": [54657375590], "values": ["L"]}]}, {"product_id": 17215385276, "sku": "12382968536834015853", "is_deleted": false, "offerings": [{"offering_id": 16982444303, "quantity": 999, "is_enabled": true, "is_deleted": false, "price": {"amount": 20.99, "divisor": 100, "currency_code": "USD"}}], "property_values": [{"property_id": 513, "property_name": "Colors", "scale_id": null, "scale_name": null, "value_ids": [50135267836], "values": ["Black"]}, {"property_id": 514, "property_name": "Sizes", "scale_id": null, "scale_name": null, "value_ids": [71931465715], "values": ["XL"]}]}, {"product_id": 16865398423, "sku": "40283645249209861078", "is_deleted": false, "offerings": [{"offering_id": 17178103946, "quantity": 999, "is_enabled": true, "is_deleted": false, "price": {"amount": 24.99, "divisor": 100, "currency_code": "USD"}}], "property_values": [{"property_id": 513, "property_name": "Colors", "scale_id": null, "scale_name": null, "value_ids": [50135267836], "values": ["Black"]}, {"property_id": 514, "property_name": "Sizes", "scale_id": null, "scale_name": null, "value_ids": [91416890176], "values": ["2XL"]}]}, {"product_id": 17215385278, "sku": "24382189257686934246", "is_deleted": false, "offerings": [{"offering_id": 17178103948, "quantity": 999, "is_enabled": true, "is_deleted": false, "price": {"amount": 28.99, "divisor": 100, "currency_code": "USD"}}], "property_values": [{"property_id": 513, "property_name": "Colors", "scale_id": null, "scale_name": null, "value_ids": [50135267836], "values": ["Black"]}, {"property_id": 514, "property_name": "Sizes", "scale_id": null, "scale_name": null, "value_ids": [94172869301], "values": ["3XL"]}]}, {"product_id": 16865398425, "sku": "31661554172351073389", "is_deleted": false, "offerings": [{"offering_id": 16982444305, "quantity": 999, "is_enabled": true, "is_deleted": false, "price": {"amount": 20.99, "divisor": 100, "currency_code": "USD"}}], "property_values": [{"property_id": 513, "property_name": "Colors", "scale_id": null, "scale_name": null, "value_ids": [52320382890], "values": ["White"]}, {"property_id": 514, "property_name": "Sizes", "scale_id": null, "scale_name": null, "value_ids": [54657375794], "values": ["S"]}]}, {"product_id": 16865398427, "sku": "24441996039451087452", "is_deleted": false, "offerings": [{"offering_id": 16982444307, "quantity": 999, "is_enabled": true, "is_deleted": false, "price": {"amount": 20.99, "divisor": 100, "currency_code": "USD"}}], "property_values": [{"property_id": 513, "property_name": "Colors", "scale_id": null, "scale_name": null, "value_ids": [52320382890], "values": ["White"]}, {"property_id": 514, "property_name": "Sizes", "scale_id": null, "scale_name": null, "value_ids": [56802856687], "values": ["M"]}]}, {"product_id": 17215385280, "sku": "61563373699021141039", "is_deleted": false, "offerings": [{"offering_id": 17178103950, "quantity": 999, "is_enabled": true, "is_deleted": false, "price": {"amount": 20.99, "divisor": 100, "currency_code": "USD"}}], "property_values": [{"property_id": 513, "property_name": "Colors", "scale_id": null, "scale_name": null, "value_ids": [52320382890], "values": ["White"]}, {"property_id": 514, "property_name": "Sizes", "scale_id": null, "scale_name": null, "value_ids": [54657375590], "values": ["L"]}]}, {"product_id": 16865398429, "sku": "62529583081829661956", "is_deleted": false, "offerings": [{"offering_id": 16982444309, "quantity": 999, "is_enabled": true, "is_deleted": false, "price": {"amount": 20.99, "divisor": 100, "currency_code": "USD"}}], "property_values": [{"property_id": 513, "property_name": "Colors", "scale_id": null, "scale_name": null, "value_ids": [52320382890], "values": ["White"]}, {"property_id": 514, "property_name": "Sizes", "scale_id": null, "scale_name": null, "value_ids": [71931465715], "values": ["XL"]}]}, {"product_id": 16865398431, "sku": "16934203286109353670", "is_deleted": false, "offerings": [{"offering_id": 16982444311, "quantity": 999, "is_enabled": true, "is_deleted": false, "price": {"amount": 24.99, "divisor": 100, "currency_code": "USD"}}], "property_values": [{"property_id": 513, "property_name": "Colors", "scale_id": null, "scale_name": null, "value_ids": [52320382890], "values": ["White"]}, {"property_id": 514, "property_name": "Sizes", "scale_id": null, "scale_name": null, "value_ids": [91416890176], "values": ["2XL"]}]}, {"product_id": 17215385282, "sku": "33064985494151884360", "is_deleted": false, "offerings": [{"offering_id": 16982444313, "quantity": 999, "is_enabled": true, "is_deleted": false, "price": {"amount": 28.99, "divisor": 100, "currency_code": "USD"}}], "property_values": [{"property_id": 513, "property_name": "Colors", "scale_id": null, "scale_name": null, "value_ids": [52320382890], "values": ["White"]}, {"property_id": 514, "property_name": "Sizes", "scale_id": null, "scale_name": null, "value_ids": [94172869301], "values": ["3XL"]}]}], "price_on_property": [513, 514], "quantity_on_property": [513, 514], "sku_on_property": [513, 514], "listing": null}`

**Comment by zkutty** [comment]
@nateswart @etsyachristensen not sure if you can help with this one?

**Comment by Gareth064** [comment]
Etsy won't help with code specific implementation issues.

There are lots of python Devs on here, hopefully someone can help you.

I'm sure I've seen other threads talking about updating pricing, you might be able to gleam from them some info on how to do it properly.

**Reply by zkutty** [reply]
Ya gotcha. I did see another thread lower which looks like a similar issue but no reply unfortunately.
